### PR TITLE
DOC: update ipython sphinxext for IPython 4.0

### PR DESCRIPTION
**Pull request by jorisvandenbossche**

### CODE REVIEW

File: doc/sphinxext/ipython_sphinxext/ipython_directive.py [hunk 1]
```diff
@@ -127,7 +127,11 @@
 from sphinx.util.compat import Directive
 
 # Our own
-from IPython import Config, InteractiveShell
+try:
+    from traitlets.config import Config
+except ImportError:
```
Comment (jreback) [inline doc/sphinxext/ipython_sphinxext/ipython_directive.py]: the fact that you have to change this is very odd. is this in their release notes? (or are we just using reallly old syntax)

Comment (TomAugspurger) [inline doc/sphinxext/ipython_sphinxext/ipython_directive.py]: They split traitlets into a new repo.

Comment (jorisvandenbossche) [inline doc/sphinxext/ipython_sphinxext/ipython_directive.py]: I don't know. Possibly they forgot to put a shim for Config (but this code is normally included in ipython itself, so then it doesn't matter. But is seems that Config is rather public object)

Comment (jreback) [inline doc/sphinxext/ipython_sphinxext/ipython_directive.py]: yeh that seems rather extreme to not to shim. I know *we* get blamed for any teeny tiny little change :)

Comment (jorisvandenbossche) [inline doc/sphinxext/ipython_sphinxext/ipython_directive.py]: They put a warning on the module (https://github.com/ipython/ipython/blob/master/IPython/config.py), but not on the object in the top-level namespace apparantly.

@takluyver is this an oversight of IPython or on purpose?

Comment (TomAugspurger) [inline doc/sphinxext/ipython_sphinxext/ipython_directive.py]: `Config` is gone entirely from the top-level namespace on IPython 4.0

Comment (takluyver) [inline doc/sphinxext/ipython_sphinxext/ipython_directive.py]: Apologies for the delayed response. I think it's an oversight that we forgot to shim for the top level module, but now that it's out there without the shim, we're probably not going to go back and add new shims, so this fix should be correct anyway.

**Pull request merged**